src/components/visualizations/chart/highcharts/getOptionalStackingConfiguration.ts

```diff
@@ -127,16 +135,25 @@ function getYAxisConfiguration(
     }
 
     const labelsVisible: IDataLabelsVisible = get<IDataLabelsVisible>(chartConfig, 'dataLabels.visible');
-    const { enabled } = getLabelsVisibilityConfig(labelsVisible);
-
-    const yAxisWithStackLabel = yAxis.map((axis: any) => ({
-        ...axis,
-        stackLabels: {
-            enabled: isNil(enabled)
-                ? true // enable by default
-                : enabled // follow dataLabels.visible config
+    const { enabled: dataLabelEnabled } = getLabelsVisibilityConfig(labelsVisible);
+
+    const yAxisWithStackLabel = yAxis.map((axis: any, index: number) => {
+        let stackLabelEnabled = isNil(dataLabelEnabled)
+            ? true // enable by default
+            : dataLabelEnabled; // follow dataLabels.visible config
+
+        if (stackMeasuresToPercent && index === 0 && hasNegativeValues(series, index)) {
```

Comment:
I think it might be better to separate this condition to a new variable:
```
const shouldDisableStackLabels = stackMeasuresToPercent && index === 0 && hasNegativeValues(series, index);
```

Reply:
Fixed